emgapi/metagenomics_exchange.py

```diff
@@ -0,0 +1,161 @@
+        return {
+            "confidence": "full",
+            "endPoint": f"https://www.ebi.ac.uk/metagenomics/analyses/{mgya}",
+            "method": ["other_metadata"],
+            "sourceID": mgya,
+            "sequenceID": run_accession,
+            "status": status,
+            "brokerID": self.broker,
+        }
+
+    def add_analysis(self, mgya: str, run_accession: str, public: bool):
+        data = self.generate_metadata(mgya, run_accession, public)
+        response = self.post_request(endpoint="datasets", data=data)
+        return response
+
+
+    def check_analysis(self, source_id: str, sequence_id: str, public=None, metadata=None) -> [str, bool]:
+        logging.info(f"Check {source_id} {sequence_id}")
+        params = {}
+        if public:
+            params = {
+                "status": "public" if public else "private",
+                "broker": self.broker
+            }
+        endpoint = f"sequences/{sequence_id}/datasets"
+        analysis_registry_id = None
+        metadata_match = True
+
+        try:
+            response = self.get_request(endpoint=endpoint, params=params)
+        except:
+            logging.warning(f"Get API request failed")
+            return analysis_registry_id , metadata_match
+
+        if response.ok:
+            data = response.json()
+            datasets = data.get("datasets")
+            sourceIDs = [item.get("sourceID") for item in datasets]
+            if source_id in sourceIDs:
+                found_record = [item for item in datasets if item.get("sourceID") == source_id][0]
+                logging.info(f"{source_id} exists in ME")
+                analysis_registry_id  = found_record.get("registryID")
+                if metadata:
+                    for metadata_record in metadata:
+                        if not(metadata_record in found_record):
+                            metadata_match = False
+                            return analysis_registry_id , metadata_match
+                        else:
+                            if metadata[metadata_record] != found_record[metadata_record]:
+                                print(metadata[metadata_record], found_record[metadata_record])
+                                metadata_match = False
+                                logging.info(f"Incorrect field {metadata[metadata_record]} != {found_record[metadata_record]})")
+                                return analysis_registry_id, metadata_match
+                return analysis_registry_id , metadata_match
+            else:
+                logging.info(f"{source_id} does not exist in ME")
+        return analysis_registry_id, metadata_match
+
+    def delete_analysis(self, registry_id: str):
+        response = self.delete_request(endpoint=f"datasets/{registry_id}")
+        if response.ok:
+            logging.info(f"{registry_id} was deleted with {response.status_code}")
+            return True
+        else:
+            if response.status_code == 400:
```

Comment:
@mberacochea perhaps we could reduce the use of conditionals here by trying something like:

error_messages = {
    400: f"Bad request for registry ID {{registry_id}}",
    401: f"Failed to authenticate for registry ID {{registry_id}}",
}

error_message = error_messages.get(response.status_code, f"Error {response.status_code})

Reply:
this also needs a `try catch`, as `delete_request` will raise an error if the response code is not in the 200 range